src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/DebtorsController.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.companieshouse.web.accounts.controller.smallfull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.debtors.Debtors;
+import uk.gov.companieshouse.web.accounts.service.smallfull.BalanceSheetService;
+import uk.gov.companieshouse.web.accounts.service.smallfull.DebtorsService;
+import uk.gov.companieshouse.web.accounts.util.Navigator;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.List;
+
+@Controller
+@NextController(ReviewController.class)
+@PreviousController(OtherAccountingPolicyController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts" +
+        "/{companyAccountsId}/small-full/debtors")
+public class DebtorsController extends BaseController {
+
+    @Autowired
+    private DebtorsService debtorsService;
+
+    @Autowired
+    private BalanceSheetService balanceSheetService;
+
+    @Override
+    protected String getTemplateName() {
+        return "smallfull/debtors";
+    }
+
+    @GetMapping
+    public String getDebtors(@PathVariable String companyNumber,
+            @PathVariable String transactionId,
+            @PathVariable String companyAccountsId,
+            Model model, HttpServletRequest request) throws ServiceException {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        try {
+            Debtors debtors = debtorsService.getDebtors(transactionId, companyAccountsId,
+                    companyNumber);
+            model.addAttribute("balanceSheet", balanceSheetService.getBalanceSheet(transactionId,
```

Comment:
You could do as I believe Jake has done and introduce a `BalanceSheetHeadings` object within the `Debtors` model. This would avoid having to add the balance sheet as a model attribute on both the get and post (which I assume is done in case any validation errors kick in and you need the dates displayed on the page?) The logic would also be a lot simpler here, and would cut down on the knowledge of the controller.